### DISCUSSION TITLE
Great work - can we count long-term on this project?

### DISCUSSION THREAD

**Discussion by allanleal** [idom-team/idom#989] [General]
@rmorshea Amazing work! Congratulations.

Can we all count on ReactPy long-term in our projects? What are your plans - is this something you see yourself working on for many years to come? Would love to get a solid yes here! :)

I'd be interested in using MaterialUI components. This will require writing Python wrappers to their JS components. Do you think this is possible to do at the moment with ReactPy?

Thanks!

**Comment by rmorshea** [comment]
I've been working on this project [since 2019](https://github.com/reactive-python/reactpy/commit/2f8087b5845c7ffac3ac42c5d5df6d122830c752) (back when it was called `idom`) and before that in a prior [ill-fated iteration from 2018](https://github.com/rmorshea/purly). I don't expect that to change any time soon.

With respect to MaterialUI, there are two main areas of friction I can foresee:

- The first is around passing children to components that are implemented in JS since the children are passed as [VDOM](https://reactpy.dev/docs/reference/specifications.html#vdom) elements rather than rendered ones. I've been thinking about some ways to make this easier, but it hasn't been high on the list of priorities since no one has asked about it. If you're interested, this is something that I could take time to work on and document.
- The second relates to the fact that (I think) in some places MUI passes components as props. This is also likely solveable, but I haven't put enough thought into it to have any concrete ideas there.

**Comment by Archmonger** [comment]
I've been helping out with this project since 2021. 

I similarly long term plans for maintenance of the following:
- [`reactpy-django`](https://github.com/reactive-python/reactpy-django)
- [Advertising](https://github.com/reactive-python/reactpy/discussions/715) and [branding](https://github.com/reactive-python/reactpy/issues/888)
- [Technical and design direction](https://github.com/reactive-python/reactpy/issues)

And upcoming tasks I'm adopting:
- [`reactpy-router`](https://github.com/reactive-python/reactpy-router)
- [`reactpy-build`](https://github.com/reactive-python/reactpy/issues/786)
- [ReactPy documentation](https://github.com/Archmonger/reactpy-docs)

**Reply by rmorshea** [reply]
Wow, someone just started making video tutorials: https://www.youtube.com/watch?v=sk1DY75ZRw0

**Comment by allanleal** [comment]
@rmorshea and @Archmonger , it's great to see so much enthusiasm from you two to make this project a success. My experience so far (one day only) with ReactPy has been quite positive. I'll try to investigate using MaterialUI with ReactPy in my spare time and tell you about my experience (and suggestions on what could be changed in ReactPy to make this process easier).

**Comment by H-G-11** [comment]
Hello there,
Would be also very curious about MUI support.
Great work, thanks a lot! Am starting to play with it.

**Reply by rmorshea** [reply]
Great to hear! Shoot us a message on [Discord](https://discord.gg/uNb5P4hA9X) if you have more questions :)

As for MUI, we have a [custom Javascript interface](https://reactpy.dev/docs/guides/escape-hatches/javascript-components.html), but we're working on [some changes](https://github.com/reactive-python/reactpy/issues/1001) that will hopefully make that much easier and streamlined.
